src/run.js

```diff
@@ -1,32 +1,38 @@
-const tsd = require('tsd');
-const { pass, fail } = require('create-jest-runner');
+// @ts-check
+
+const tsd = require("tsd");
+const { pass } = require("create-jest-runner");
+const { fail } = require("./fail");
 
 module.exports = async ({ testPath }) => {
   // Convert absolute path to relative path
-  const testFile = testPath.replace(process.cwd(), '');
+  const testFile = testPath.replace(process.cwd(), "");
   const start = +new Date();
-  const diagnose = await tsd({
+  const diagnose = await tsd.default({
     cwd: process.cwd(),
-    testFiles: [ testFile ]
+    testFiles: [testFile],
   });
 
-  // Currently only reports a single error.
   if (diagnose.length > 0) {
+    const failures = diagnose.map((test) => ({
+      path: test.fileName,
+      errorMessage: test.message,
+      title: `${testFile}:${test.line}:${test.column} - ${test.severity} - ${test.message}`
```

Comment:
@SaurabhAgarwala let me know if this seems like a good title for you

Reply:
Yeah this looks good